File: Assets/Scripts/graphics/csharp/ShotLight.cs
using UnityEngine;
using System.Collections;

public class ShotLight : MonoBehaviour {

	public float time = 0.02f;
	private float timer;
	
	void OnEnable()
	{
		if (GetComponent<Light>() == null)
		{
			Destroy(this);
		}
		else
		{
			timer = time;
			GetComponent<Light>().enabled = false;
		}
	}

	void OnDisable()
	{
		if (GetComponent<Light>() == null)
		{
			Destroy(this);
		}
		else
		{
			timer = time;
			GetComponent<Light>().enabled = false;
		}
	}

	void LateUpdate()
	{
		timer -= Time.deltaTime;

		if (timer <= 0.0)
		{
			timer = time;
			GetComponent<Light>().enabled = !GetComponent<Light>().enabled;
		}
	}
}

File: Assets/Scripts/graphics/ShotLight.js
#pragma strict
#pragma implicit
#pragma downcast

class ShotLight extends MonoBehaviour
{
	public var time : float = 0.02;
	private var timer : float;
	
	function OnEnable()
	{
		if(GetComponent.<Light>() == null)
		{
			Destroy(this);
		}
		else
		{
			timer = time;
			GetComponent.<Light>().enabled = false;
		}
	}
	
	function OnDisable()
	{
		if(GetComponent.<Light>() == null)
		{
			Destroy(this);
		}
		else
		{
			timer = time;
			GetComponent.<Light>().enabled = false;
		}
	}
	
	function LateUpdate()
	{
		timer -= Time.deltaTime;
		
		if(timer <= 0.0)
		{
			timer = time;
			GetComponent.<Light>().enabled = !GetComponent.<Light>().enabled;
		}
	}
}